### ISSUE TITLE
match_same_arms: false negative

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#4244]
````rust

#![deny(clippy::match_same_arms)]

#[derive(PartialEq, PartialOrd, Eq, Ord)]
pub enum CommandInfo {
    BuiltIn { name: String, about: Option<String> },
    External { name: String, path: std::path::PathBuf },
}

impl CommandInfo {
    pub fn name(&self) -> String {
        match self {
            CommandInfo::BuiltIn { name, .. } => name.to_string(),
            CommandInfo::External { name, .. } => name.to_string(),
        }
    }
}

fn main() {}
````
This used to trigger back on June 3rd but no longer does in July 1st (`clippy 0.0.212 (e3cb40e 2019-06-25)`)

Looks like a false negative to me?

---
EDIT(@flip1995): [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=153640b8c5fd0ef541a9227a07dc8d94)

EDIT(matthiaskrgr): code is from cargo 0.35.0
